prov/verbs/src/verbs_info.c

```diff
@@ -967,15 +967,38 @@ static void fi_ibv_sockaddr_set_port(struct sockaddr *sa, uint16_t port)
 	}
 }
 
-static int fi_ibv_fill_addr(uint64_t flags, struct rdma_addrinfo *rai,
-		struct fi_info **info, struct rdma_cm_id *id)
+static int fi_ibv_is_wildcard_addr(struct sockaddr *sa)
+{
+	struct sockaddr_in *sa_in;
+	struct sockaddr_in6 *sa_in6;
+	struct in_addr ia_any = {
+		.s_addr = INADDR_ANY,
+	};
+	struct in6_addr ia6_any = IN6ADDR_ANY_INIT;
+
+	switch(sa->sa_family) {
+	case AF_INET:
+		sa_in = (struct sockaddr_in *)sa;
+		return !memcmp(&sa_in->sin_addr.s_addr, &ia_any,
+				sizeof(ia_any)) && sa_in->sin_port ? 1 : 0;
+	case AF_INET6:
+		sa_in6 = (struct sockaddr_in6 *)sa;
+		return !memcmp(&sa_in6->sin6_addr, &ia6_any,
+				sizeof(ia6_any)) && sa_in6->sin6_port ? 1 : 0;
+	default:
+		FI_WARN(&fi_ibv_prov, FI_LOG_CORE, "Unknown address format!\n");
+		return 0;
+	}
+}
+
+static int fi_ibv_fill_addr(struct rdma_addrinfo *rai, struct fi_info **info,
+		struct rdma_cm_id *id)
 {
 	struct fi_info *fi;
 	struct sockaddr *local_addr;
 	int ret;
 
-	/* Just FI_SOURCE would correspond to a wildcard address */
-	if (id->verbs || (flags & FI_SOURCE)) {
+	if (id->verbs || fi_ibv_is_wildcard_addr(rai->ai_src_addr)) {
```

Comment:
Since, there is a check `if (id->verbs && !rai->ai_src_addr) {` below, I guess, we need to handle two cases separately:
1. rai->ai_src_addr == NULL
2. rai->ai_src_addr != NULL